Build a regression model.

In [18]:
# import numpy
import numpy as np
import pandas as pd
# import linear_model and datasets from sklearn
from sklearn import linear_model, datasets
from sklearn.datasets import fetch_california_housing

In [22]:
pd.read_csv('yelp_merge.csv')
yelp_merge = pd.read_csv('yelp_merge.csv')
yelp_merge = yelp_merge.drop('Unnamed: 0', axis=1)
yelp_merge

name_x  searched_longitude  searched_latitude  \
0          WESTERN WAY           -6.269425          53.354929   
1          WESTERN WAY           -6.269425          53.354929   
2          WESTERN WAY           -6.269425          53.354929   
3          WESTERN WAY           -6.269425          53.354929   
4          WESTERN WAY           -6.269425          53.354929   
...                ...                 ...                ...   
1135  KILLARNEY STREET           -6.247579          53.354845   
1136  KILLARNEY STREET           -6.247579          53.354845   
1137  KILLARNEY STREET           -6.247579          53.354845   
1138  KILLARNEY STREET           -6.247579          53.354845   
1139  KILLARNEY STREET           -6.247579          53.354845   

      number of bikes                     name_y              location  \
0                  40            The Boxty House      20-21 Temple Bar   
1                  40    The Old Mill Restaurant  14 Temple Bar Square   
2                  40   Bleecker Street Cafe Bar      68 Dorset Street   
3                  40               IL Vicoletto         5 Crow Street   
4                  40  The Bank on College Green   20-22 College Green   
...               ...                        ...                   ...   
1135               30               Featherblade     51B Dawson Street   
1136               30              Darkey Kellys         Hotel Harding   
1137               30                F.X.Buckley         2 Crow Street   
1138               30                     Pichet  14-15 Trinity Street   
1139               30                     Bunsen     22 Essex Street E   

      rating     distance  review_count price  
0        4.3  1130.342156          1231   €€€  
1        4.5  1100.844806           229    €€  
2        4.6   421.257967            58    €€  
3        4.5  1218.680265           131   €€€  
4        4.4  1294.204466           441   €€€  
...      ...          ...           ...   ...  
1135     4.4  1632.667366           122    €€  
1136     4.6  1926.342791           263    €€  
1137     4.6  1569.462456           120   €€€  
1138     4.3  1547.853602           116   €€€  
1139     4.5  1523.550039           215    €€  

[1140 rows x 10 columns]

In [37]:
y = yelp_merge['rating']
x = yelp_merge[['searched_latitude', 'searched_longitude', 'review_count']]

In [39]:
import statsmodels.api as sm

In [41]:
X = sm.add_constant(x) # adding a constant
lin_reg = sm.OLS(y,X)

In [45]:
model = lin_reg.fit()

Provide model output and an interpretation of the results. 

In [47]:
print_model = model.summary()
print(print_model)

                            OLS Regression Results                            
Dep. Variable:                 rating   R-squared:                       0.017
Model:                            OLS   Adj. R-squared:                  0.015
Method:                 Least Squares   F-statistic:                     6.727
Date:                Mon, 05 May 2025   Prob (F-statistic):           0.000169
Time:                        19:34:53   Log-Likelihood:                 142.25
No. Observations:                1140   AIC:                            -276.5
Df Residuals:                    1136   BIC:                            -256.3
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
                         coef    std err          t      P>|t|      [0.025      0.975]
--------------------------------------------------------------------------------------
const               -103.1681     44

R-squared shows that the model explains a very small, only 1.7% of the variation in the results. 
p values shows that every independent variable is statistically significant as related to the rating, although very weak.

# Stretch

How can you turn the regression model into a classification model?